### PR TITLE
Move to GitHub Actions

### PR DESCRIPTION
**Pull request by mseri**
Signed-off-by: Marcello Seri <marcello.seri@gmail.com>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          ocaml-version: ${{ matrix.ocaml-version }}
+          opam-depext: false
+          opam-pin: false
+
+      - run: |
+          opam pin add owl-ode-base.dev . --no-action
+          opam pin add owl-ode.dev . --no-action
+          opam pin add owl-ode-odepack.dev . --no-action
+
+      - run: opam depext --yes --with-test owl-ode-base owl-ode owl-ode-odepack
```
Comment (kit-ty-kate) [inline .github/workflows/ci.yml]:
```suggestion
      - run: |
          brew list
          opam depext --yes --with-test owl-ode-base owl-ode owl-ode-odepack
          brew list
          ls /usr/local/bin
          gfortran --version
          which gfortran
```
could you add this temporarily for testing purpose?

Comment (mseri) [reply .github/workflows/ci.yml]: Sure thing!

**Comment by mseri** [comment]
Now only one of the three worked… weird

@smorimoto is it possible that it is related to the master version of setup-ocaml that I am using? It is the one you were testing on cohttp some time ago

**Comment by smorimoto** [comment]
It's weird. Let me check it out later. By the way, we plan to move avsm/setup-ocaml under ocaml org and release v2 this week (or next week), so it's better to use that. (See https://github.com/avsm/setup-ocaml/pull/66)

**Comment by mseri** [comment]
Also v2 fails without finding gfortran (the output is a bit weirder) but it is less predictable. After two re-runs it is no longer failing

**Comment by smorimoto** [comment]
Well, I was late for the boat, but I generally agree with cannorin.

**Comment by smorimoto** [comment]
I guess this was a problem with the GitHub Actions runner. (I just put a comment on this thread that I thought I wrote on another thread! Just disregard it hehe)

**Pull request merged**